### ISSUE TITLE
Add support for file source

### ISSUE THREAD

**Issue by POPSuL** [ApiGen/ApiGen#930]
Now we can't provide file as source, and...

```
% php ./bin/apigen generate -vvv --destination=/var/www/api /var/www/core/src /var/www/core/system.php /var/www/core/exceptions.php

                                                                     
  [Roave\BetterReflection\SourceLocator\Exception\InvalidDirectory]  
  "/var/www/core/system.php" must be a directory, not a file         
                                                                     

Exception trace:
 () at /home/popsul/projects/ApiGen/vendor/roave/better-reflection/src/SourceLocator/Exception/InvalidDirectory.php:18
 Roave\BetterReflection\SourceLocator\Exception\InvalidDirectory::fromNonDirectory() at /home/popsul/projects/ApiGen/vendor/roave/better-reflection/src/SourceLocator/Type/DirectoriesSourceLocator.php:39
 Roave\BetterReflection\SourceLocator\Type\DirectoriesSourceLocator->Roave\BetterReflection\SourceLocator\Type\{closure}() at n/a:n/a
 array_map() at /home/popsul/projects/ApiGen/vendor/roave/better-reflection/src/SourceLocator/Type/DirectoriesSourceLocator.php:47
 Roave\BetterReflection\SourceLocator\Type\DirectoriesSourceLocator->__construct() at /home/popsul/projects/ApiGen/packages/Reflection/src/Parser/Parser.php:121
 ApiGen\Reflection\Parser\Parser->createDirectoriesSource() at /home/popsul/projects/ApiGen/packages/Reflection/src/Parser/Parser.php:52
 ApiGen\Reflection\Parser\Parser->parseDirectories() at /home/popsul/projects/ApiGen/src/Application/ApiGenApplication.php:55
 ApiGen\Application\ApiGenApplication->runCommand() at /home/popsul/projects/ApiGen/src/Console/Command/GenerateCommand.php:61
 ApiGen\Console\Command\GenerateCommand->execute() at /home/popsul/projects/ApiGen/vendor/symfony/console/Command/Command.php:264
 Symfony\Component\Console\Command\Command->run() at /home/popsul/projects/ApiGen/vendor/symfony/console/Application.php:869
 Symfony\Component\Console\Application->doRunCommand() at /home/popsul/projects/ApiGen/vendor/symfony/console/Application.php:223
 Symfony\Component\Console\Application->doRun() at /home/popsul/projects/ApiGen/vendor/symfony/console/Application.php:130
 Symfony\Component\Console\Application->run() at /home/popsul/projects/ApiGen/bin/apigen:16

generate [--destination DESTINATION] [--] <source> (<source>)...
```

Also need `--exclude` option.

**Comment by TomasVotruba** [comment]
Could you add file option?

It would require:

- add `private parseFiles(array $files)` method to [`Parser`](https://github.com/ApiGen/ApiGen/blob/d302650ae99fc796e223831131afce231def825f/packages/Reflection/src/Parser/Parser.php#L47-L58) + `public parseFilesAndDirectories(array $sources)` method above them
- make `parseDirectories(array $directories)` private
- use `public parseFilesAndDirectories(array $sources)` where `parseDirectories(array $directories)` is used
- update [`SourceOption`](https://github.com/ApiGen/ApiGen/blob/d302650ae99fc796e223831131afce231def825f/packages/ModularConfiguration/src/Option/SourceOption.php#L39)
- add [`ParserTest`](https://github.com/ApiGen/ApiGen/tree/master/packages/Reflection/tests/Parser) here to test all 3 methods

As for `--exclude` option, please open new issue after this is finished. I got different idea for that.

**Comment by vlastavesely** [comment]
Could be closed after #944 got merged. What do you plan with the `--exclude`?

**Comment by POPSuL** [comment]
Added issue #947

**Comment by TomasVotruba** [comment]
Thanks, let's continue there
